### DISCUSSION TITLE
Expose pREST directly to the Internet?

### DISCUSSION THREAD

**Discussion by MichaelHipp** [prest/prest#132] [API Server] [answered]
Question:  can pREST be sufficiently secured to make it safe to directly expose it to the Internet? Thanks.

**Comment by pablomelo** [comment, marked as answer]
I would say that it depends on what "sufficiently secured" means for each user.

There are a few security techniques/features that can be applied that would cover the majority of cases:

- JWT - Json Web Tokens
- Strict CORS to make sure the data doesn't leak
- And https end-to-end to make sure no one can intercept or change the data while in transit.

Still, that would not be "sufficient" for regulatory compliance required for financial, health and government data.

There's a trade-off between security and convenience which the balance between both has to be determined through analyzing the risk and consequence of unauthorized data access.

I use PREST on more than one business application. On these applications, only employees are granted access and with this, we get the security of the "walled-garden" and the flexibility of direct access to Postgres on the browser.

I'm not sure if that answers your question, but I hope that is a starting point.
